pkg/api/v20170701/types.go

```diff
@@ -0,0 +1,294 @@
+	KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
+	// caPrivateKey is an internal field only set if generation required
+	caPrivateKey string
+}
+
+// LinuxProfile represents the Linux configuration passed to the cluster
+type LinuxProfile struct {
+	AdminUsername string `json:"adminUsername"`
+
+	SSH struct {
+		PublicKeys []struct {
+			KeyData string `json:"keyData"`
+		} `json:"publicKeys"`
+	} `json:"ssh"`
+	Secrets []KeyVaultSecrets `json:"secrets,omitempty"`
+}
+
+// WindowsProfile represents the Windows configuration passed to the cluster
+type WindowsProfile struct {
+	AdminUsername string            `json:"adminUsername,omitempty"`
+	AdminPassword string            `json:"adminPassword,omitempty"`
+	Secrets       []KeyVaultSecrets `json:"secrets,omitempty"`
+}
+
+// ProvisioningState represents the current state of container service resource.
+type ProvisioningState string
+
+const (
+	// Creating means ContainerService resource is being created.
+	Creating ProvisioningState = "Creating"
+	// Updating means an existing ContainerService resource is being updated
+	Updating ProvisioningState = "Updating"
+	// Failed means resource is in failed state
+	Failed ProvisioningState = "Failed"
+	// Succeeded means resource created succeeded during last create/update
+	Succeeded ProvisioningState = "Succeeded"
+	// Deleting means resource is in the process of being deleted
+	Deleting ProvisioningState = "Deleting"
+	// Migrating means resource is being migrated from one subscription or
+	// resource group to another
+	Migrating ProvisioningState = "Migrating"
+)
+
+// OrchestratorProfile contains Orchestrator properties
+type OrchestratorProfile struct {
+	OrchestratorType    OrchestratorType    `json:"orchestratorType"`
+	OrchestratorVersion OrchestratorVersion `json:"orchestratorVersion"`
+}
+
+// MasterProfile represents the definition of master cluster
+type MasterProfile struct {
+	Count                    int    `json:"count"`
+	DNSPrefix                string `json:"dnsPrefix"`
+	VMSize                   string `json:"vmSize"`
+	OSDiskSizeGB             int    `json:"osDiskSizeGB,omitempty"`
+	VnetSubnetID             string `json:"vnetSubnetID,omitempty"`
+	FirstConsecutiveStaticIP string `json:"firstConsecutiveStaticIP,omitempty"`
+	StorageProfile           string `json:"storageProfile,omitempty"`
+
+	// subnet is internal
+	subnet string
+	// Master LB public endpoint/FQDN with port
+	// The format will be FQDN:2376
+	// Not used during PUT, returned as part of GET
+	FQDN string `json:"fqdn,omitempty"`
+}
+
+// AgentPoolProfile represents configuration of VMs running agent
+// daemons that register with the master and offer resources to
+// host applications in containers.
+type AgentPoolProfile struct {
+	Name           string `json:"name"`
+	Count          int    `json:"count"`
+	VMSize         string `json:"vmSize"`
+	OSDiskSizeGB   int    `json:"osDiskSizeGB,omitempty"`
```

Comment:
The PR description mentions: "Multiple agents pools with public port configuration and choice of OsType"
Any reason why OSDiskSizeGB , StorageProfile and VnetSubnetID  is being exposed in the public API model?